### PR TITLE
Add env vars for osx-arm64 cross-compile

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -392,6 +396,17 @@ Mingw_w64_makefiles() {
 Darwin() {
     unset JAVA_HOME
 
+    if [[ $CONDA_BUILD_CROSS_COMPILATION == 1 && $target_platform == osx-arm64 ]]; then
```
Comment (isuruf) [inline recipe/build.sh:399]:
```suggestion
    if [[ $CONDA_BUILD_CROSS_COMPILATION == 1 ]]; then
```

Comment (isuruf) [reply recipe/build.sh:399]: Can you also move this block out of `Darwin()` function and run it for both `linux` and `osx` (not `win`)?

Comment (erykoff) [reply recipe/build.sh:399]: Do I create a function to do that? Or copy and paste it twice?

Comment (isuruf) [reply recipe/build.sh:399]: You can do it at the beginning of this file or create a function

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by erykoff** [comment]
The "allow edits by maintainers" box is checked.

**Pull request merged**